### ISSUE TITLE
There are some issues, guessing with the generator

### ISSUE THREAD

**Issue by sudokit** [foxik0169/imgui-odin#3]
I havent really looked deeper into it, but some errors popped up when importing imgui-odin library and trying to build any project. Some issues including:
- using the keyword `context` as a function argument, which is a reserved keyword
... so on :p
Might just be something with a newer odin version, but though it just drop this here
odin version is: `odin version dev-2024-06:ff2d04231`

**Comment by foxik0169** [comment]
Hey, yeah probably odin changed a littlebit since this generator has been updated. Should not be that hard to fix. I am open to pull requests if you find solution faster than me.

Dan

**Comment by sudokit** [comment]
Hey sorry forgot about this :p yeah i can try to fiddle around and see if id get it to work. Really not too familiar with generators yet, but ill try!

**Comment by sudokit** [comment]
Got some progress, but damn is it broken :p made a weird hacky fix.sh script to fix some things that i could figure out on how to patch the generator to do them

**Comment by foxik0169** [comment]
Yeah, I would go about it in a similar way I think. Or maybe just change the generator by hand... have no idea what changed since then so I have less context than you.

**Comment by foxik0169** [comment]
Yeah it seems that imgui itself and also odin changed a lot. So the generator doesn't handle those cases. There are some obvious ones that can be fixed easily (for example handling type mapping of "signed char" to "i8") but I would have to spend some time doing that which I can not afford now. 

You don't have to do external fix scripts, you can update the generator to support new syntax. If you still wan't to use this. I am not planing to use odin anytime soon so I don't think I will take a look at this in near future. _

Last working cimgui I used with this was hash: b0649485e96c3b25e8b691aca489c89f9913e53a on branch docking inter.
Might start with that.

**Comment by sudokit** [comment]
On the other hand https://gitlab.com/L-4/odin-imgui seems to work really really well... maybe it could be used for now

**Comment by foxik0169** [comment]
Yeah might be in better shape.
